src/coreclr/vm/jitinterface.cpp

```diff
@@ -10722,12 +10733,15 @@ void* CEEJitInfo::getHelperFtn(CorInfoHelpFunc    ftnNum,         /* IN  */
                     {
                         // Cache it for future uses to avoid taking the lock again.
                         hlpFinalTierAddrTable[dynamicFtnNum] = finalTierAddr;
+                        EE_TO_JIT_TRANSITION_LEAF();
```

Comment:
There is one more return earlier in the method that should get the same treatment.